src/App/Document.cpp

```diff
@@ -3571,15 +3570,8 @@ DocumentObject* Document::addObject(const char* sType,
     }
 
     // get Unique name
-    string ObjectName;
-
-    if (pObjectName && pObjectName[0] != '\0') {
-        ObjectName = getUniqueObjectName(pObjectName);
-    }
-    else {
-        ObjectName = getUniqueObjectName(sType);
-    }
-
+    const bool hasName = pObjectName && pObjectName[0] != '\0';

```

Comment:
In another PR i remarked that we really need something like `Base::isEmptyString(...)` - this is yet another case.